### PR TITLE
matrix-dendrite: rename to dendrite

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
No other distro calls it matrix-dendrite: https://repology.org/project/matrix-dendrite.
They call it just dendrite: https://repology.org/project/dendrite.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by infinisil — APPROVED** [review]

**Comment by mjlbach** [comment]
This is fine, I only named it matrix-dendrite to match synapse.

**Comment by mjlbach** [comment]
I thought the matrix maintainers was taggable, but this should also probably get the signoff of some people from that maintainer group that commented on the previous PR.

**Comment by dotlambda** [comment]
> I thought the matrix maintainers was taggable, but this should also probably get the signoff of some people from that maintainer group that commented on the previous PR.

All Matrix maintainers were pinged by ofborg.

File: nixos/modules/services/misc/dendrite.nix
Comment (sumnerevans) [inline nixos/modules/services/misc/dendrite.nix:6]: Is this a breaking change? Is there manual intervention necessary to move the certs and such?

Comment (dotlambda) [reply nixos/modules/services/misc/dendrite.nix:6]: Theoretically not, because the paths to `tlsKey` etc. have to be specified manually. However, this will practically break configuration because the module is broken in a way that forces the keys to be within the `workingDir`.
Oh and user data will have to be moved manually as well.

Anyway, I think this is fine because the module has only been around for a few days.

**Comment by SuperSandro2000** [comment]
The package name on debian will most likely be dendrite because it is free unlike synapse.

**Comment by mjlbach** [comment]
If you all feel confident that RHEL/Debian will use `dendrite` over `matrix-dendrite`, then this is fine with me. I just didn't want to base this decision on the `AUR`/`alpine`...

**Comment by dotlambda** [comment]
I'm merging this so we have time to merge the necessary fixes for the service before 21.05 is branched off.

**Pull request merged**
